packages/patternfly-4/react-table/src/components/Table/base/body-row.tsx

```diff
@@ -1,29 +1,40 @@
 /**
- * body-row.js
+ * body-row.tsx
  *
  * Forked from reactabular-table version 8.14.0
  * https://github.com/reactabular/reactabular/tree/v8.14.0/packages/reactabular-table/src
- * */
-
+ */
 import { isEqual, isFunction } from 'lodash';
-import React from 'react';
-import columnsAreEqual from './columns-are-equal';
-import evaluateFormatters from './evaluate-formatters';
-import evaluateTransforms from './evaluate-transforms';
-import mergeProps from './merge-props';
-import { tableBodyRowDefaults, tableBodyRowTypes } from './types';
+import * as React from 'react';
+import { columnsAreEqual } from './columns-are-equal';
+import { evaluateFormatters } from './evaluate-formatters';
+import { evaluateTransforms } from './evaluate-transforms';
+import { mergeProps } from './merge-props';
+import { createElementType, formatterValueType, ColumnType, RowType, RenderersTypes } from './types';
+
+export interface BodyRowProps {
+  columns: RenderersTypes['columns'];
+  renderers: RenderersTypes['renderers']['body'];
+  onRow?: Function;
+  rowIndex: number;
+  rowData: RowType;
+  rowKey: string;
+}
 
-class BodyRow extends React.Component {
-  shouldComponentUpdate(nextProps) {
-    // eslint-disable-line no-unused-vars
+export class BodyRow extends React.Component<BodyRowProps, {}> {
+  static defaultProps = {
+    onRow: (...args:any) => Object 
+  };
+
+  shouldComponentUpdate(nextProps: BodyRowProps){
     const previousProps = this.props;
 
     // Check for row based override.
     const { renderers } = nextProps;
 
-    if (renderers && renderers.row && renderers.row.shouldComponentUpdate) {
-      if (isFunction(renderers.row.shouldComponentUpdate)) {
-        return renderers.row.shouldComponentUpdate.call(this, nextProps);
+    if (renderers && renderers.row && (renderers.row as React.Component).shouldComponentUpdate) {
+      if (isFunction((renderers.row as React.Component).shouldComponentUpdate)) {
+        return (renderers.row as React.Component).shouldComponentUpdate.call(this, nextProps, {}, {});
```

Comment:
This is really weird and sort of not how `shouldComponentUpdate` is supposed to work, but I'm not going to question it because it was preexisting :)

Reply:
Yes.... yes it is weird.... 

This is all existing logic in Reactabular. FWIW, I don't think calling `shouldComponentUpdate` on a consumer passed component is good React practice, but that is seen as an extendability "feature" from the Reactabular source... 

It also makes typing really hard... I'm OK w/ removing this if others don't see the need to use it. It could help on some edge cases where you are extending body-row though...

Reply:
Probably better ways to write this in the future...

Reply:
Let's leave it for now, I'm sure it's black magic that is necessary somehow and we'll probably think of a better way eventually.